File: src/main/java/ro/nicolaemariusghergu/queryit/repository/ManufacturerRepository.java
package ro.nicolaemariusghergu.queryit.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.lang.NonNull;
import org.springframework.stereotype.Repository;
import ro.nicolaemariusghergu.queryit.model.Manufacturer;

import java.util.List;
import java.util.Optional;

@Repository
public interface ManufacturerRepository extends JpaRepository<Manufacturer, Long> {

    @NonNull
    @Override
    Optional<Manufacturer> findById(@NonNull Long id);

    @NonNull
    @Override
    List<Manufacturer> findAll();

    Optional<Manufacturer> findByName(String name);

    @NonNull
    @Override
    <S extends Manufacturer> S save(@NonNull S entity);

    @NonNull
    @Override
    <S extends Manufacturer> List<S> saveAll(@NonNull Iterable<S> entities);

}

File: src/main/java/ro/nicolaemariusghergu/queryit/repository/ManufacturerRepository.kt
package ro.nicolaemariusghergu.queryit.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.lang.NonNull
import org.springframework.stereotype.Repository
import org.springframework.transaction.annotation.Transactional
import ro.nicolaemariusghergu.queryit.model.Manufacturer
import java.util.*

@Repository
@Transactional
interface ManufacturerRepository : JpaRepository<Manufacturer?, Long?> {
    @NonNull
    override fun findById(@NonNull id: Long?): Optional<Manufacturer?>?
    @NonNull
    override fun findAll(): MutableList<Manufacturer?>?
    open fun findByName(name: String?): Optional<Manufacturer?>?
    @NonNull
    override fun <S : Manufacturer?> save(@NonNull entity: S?): S?
    @NonNull
    override fun <S : Manufacturer?> saveAll(@NonNull entities: Iterable<S?>?): MutableList<S?>?
}